Apps/GatewayApi/src/Services/UserEmailService.cs

```diff
@@ -241,8 +257,8 @@ private void AddVerificationEmail(string hdid, string toEmail, Guid inviteKey, b
             if (isVerified)
             {
                 messageVerification.Email.EmailStatusCode = EmailStatus.Processed;
-                this.messageVerificationDelegate.Insert(messageVerification);
-                this.ValidateEmail(hdid, inviteKey);
+                this.messageVerificationDelegate.Insert(messageVerification, commit);
+                this.ValidateEmail(hdid, inviteKey).GetAwaiter();
```

Comment:
Seems like this should be an await and the method should be made async as well?

Reply:
This has far flung consequences and this was the smallest impact. This is definitely an improvement and Can add a ticket for improving async code for services.

I don't want to change this now due to the extended scope already, the original task didn't include the notification channel in the description, so this already increased scope and my first step was to do exactly what you suggested however the amount of calls to this function would see a numerous functions needing to be updated to async (cannot remember the exact number).

Reply:
Can we merge this to a feature branch instead then, so it can be fixed before merging to dev?

Reply:
I don't think that's a good approach the feature is essentially deliverable and this would fall under the optimization category, I would have it as a separate task/story to introduce asynchronous code through the "{X}Controller" paths.